web/packages/api/src/toEthereum_v2.ts

```diff
@@ -481,22 +554,36 @@ export async function validateTransfer(
 ): Promise<ValidationResult> {
     const { sourceParachain, gateway, bridgeHub, assetHub } = connections
     const { registry, fee, tokenAddress, amount, beneficiaryAccount } = transfer.input
-    const { sourceAccountHex, sourceParaId, sourceParachain: source } = transfer.computed
+    const {
+        sourceAccountHex,
+        sourceParaId,
+        sourceParachain: source,
+        sourceAssetMetadata,
+    } = transfer.computed
     const { tx } = transfer
 
     const logs: ValidationLog[] = []
 
-    const [nativeBalance, dotBalance, tokenBalance] = await Promise.all([
+    const [nativeBalance, dotBalance] = await Promise.all([
         getNativeBalance(sourceParachain, sourceAccountHex),
         getDotBalance(sourceParachain, source.info.specName, sourceAccountHex),
-        getTokenBalance(
+    ])
+    let tokenBalance: any
+    if (
+        transfer.computed.ahAssetMetadata.location?.parents == DOT_LOCATION.parents &&
```

Comment:
Maybe we should use the token address for DOT for this. I am not sure if this will match every version of xcm. We need to find a way to normalize multi-locations. I think in a separate PR we need to have a runtime API on bridgehub which takes in a versioned multi-location and returns the token_id. And when we index the assets on each chain, we normalize it to the token_id and use the token id as a stable id for multi-locations.

Reply:
> have a runtime API on bridgehub which takes in a versioned multi-location and returns the token_id

This branch is only for transferring `DOT` from AH, which is a special case that retrieves the token balance from the `Balance` pallet. I also tested that the `{ parents: 1, interior: "Here" }` representation works for both XCMV4 and XCMV5. So, I prefer to keep things simple here.